Agrume/Agrume.swift

```diff
@@ -253,13 +262,30 @@ public final class Agrume: UIViewController {
   /// - Parameters:
   ///   - index: The target index
   ///   - url: The replacement URL
-  public func updateImage(at index: Int, with url: URL) {
+  ///   - newTitle: The new title, if nil then no change
+  public func updateImage(at index: Int, with url: URL, newTitle: NSAttributedString? = nil) {
     assert(images.count > index)
-    let replacement = with(images[index]) { $0.url = url }
+    let replacement = with(images[index]) {
+      $0.image = nil
+      $0.url = url
+      if let newTitle = newTitle {
+        $0.title = newTitle
+      }
+    }
+    
+    markAsUpdatingSameCell(at: index)
     images[index] = replacement
     reload()
   }
   
+  private func markAsUpdatingSameCell(at index: Int) {
+    collectionView.visibleCells.forEach({ cell in
+      if let cell = cell as? AgrumeCell, cell.index == index {
+        cell.updatingImageOnSameCell = true
+      }
+    })
```

Comment:
nit: prefer trailing closure syntax
```suggestion
    collectionView.visibleCells.forEach { cell in
      if let cell = cell as? AgrumeCell, cell.index == index {
        cell.updatingImageOnSameCell = true
      }
    }
```